Auto update failing for rgbds

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the rgbds package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="bash">
INFO: Updating rgbds [Current version: 0.5.2]
INFO: package being updated to 0.6.0.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 7853k  100 7853k    0     0  12.0M      0 --:--:-- --:--:-- --:--:-- 12.0M

You are about to commit these changes:

--------------------
diff --git a/packages/rgbds/build.sh b/packages/rgbds/build.sh
index 9140c142b..c535c5f19 100644
--- a/packages/rgbds/build.sh
+++ b/packages/rgbds/build.sh
@@ -2,8 +2,8 @@ TERMUX_PKG_HOMEPAGE=https://rgbds.gbdev.io
 TERMUX_PKG_DESCRIPTION="Rednex Game Boy Development System - An assembly toolchain for the Nintendo Game Boy & Game Boy Color"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=0.5.2
+TERMUX_PKG_VERSION="0.6.0"
 TERMUX_PKG_SRCURL=https://github.com/gbdev/rgbds/releases/download/v${TERMUX_PKG_VERSION}/rgbds-${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=142479e1b68e3947af2dace884fb0efa65e789163cb5909bdcbae1f132dc1457
+TERMUX_PKG_SHA256=a5fa40b92e0562c6a092bc81ca56614316730aca7b10b8177285023781ff0e32
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libpng"
--------------------

rgbds: update to 0.6.0


Not committing to Git!

INFO: Trying to build package.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building rgbds for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13133  100 13133    0     0  42660      0 --:--:-- --:--:-- --:--:-- 42778
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   3076      0 --:--:-- --:--:-- --:--:--  3085
gpg: Signature made Sun 02 Oct 2022 02:30:30 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13319  100 13319    0     0  39545      0 --:--:-- --:--:-- --:--:-- 39522
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   3039      0 --:--:-- --:--:-- --:--:--  3051
gpg: Signature made Sun 04 Sep 2022 06:32:25 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13119  100 13119    0     0  48938      0 --:--:-- --:--:-- --:--:-- 49134
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2926      0 --:--:-- --:--:-- --:--:--  2922
100   833  100   833    0     0   2925      0 --:--:-- --:--:-- --:--:--  2922
gpg: Signature made Sun 02 Oct 2022 05:46:36 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading dependency zlib@1.2.12-1 if necessary...
Found zlib in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/z/zlib/zlib_1.2.12-1_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 59984  100 59984    0     0   289k      0 --:--:-- --:--:-- --:--:--  291k
extracting zlib to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency libpng@1.6.38 if necessary...
Found libpng in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libp/libpng/libpng_1.6.38_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  190k  100  190k    0     0  1083k      0 --:--:-- --:--:-- --:--:-- 1085k
extracting libpng to /home/builder/.termux-build/_cache-aarch64...
Downloading https://github.com/gbdev/rgbds/releases/download/v0.6.0/rgbds-0.6.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 7853k  100 7853k    0     0  9535k      0 --:--:-- --:--:-- --:--:-- 9535k
Applying patch: src-extern-getopt.c.patch
Downloading https://cmake.org/files/v3.24/cmake-3.24.2-linux-x86_64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 44 44.8M   44 20.0M    0     0  20.9M      0  0:00:02 --:--:--  0:00:02 20.9M
100 44.8M  100 44.8M    0     0  32.6M      0  0:00:01  0:00:01 --:--:-- 32.7M
Downloading https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  100k  100  100k    0     0   215k      0 --:--:-- --:--:-- --:--:--  215k
Archive:  /home/builder/.termux-build/rgbds/tmp/ninja-1.10.2.zip
  inflating: /home/builder/.termux-build/_cache/ninja-1.10.2/ninja  
-- Android: Targeting API '24' with architecture 'arm64', ABI 'arm64-v8a', and processor 'aarch64'
-- The C compiler identification is Clang 14.0.6
-- The CXX compiler identification is Clang 14.0.6
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/builder/.termux-build/_cache/android-r25b-api-24-v0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/builder/.termux-build/_cache/android-r25b-api-24-v0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- RGBDS version: 
-- Found PkgConfig: /home/builder/.termux-build/_cache/android-r25b-api-24-v0/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'libpng'
--   Found libpng, version 1.6.38
-- Found BISON: /usr/bin/bison (found version "3.8.2") 
-- Looking for sin in m
-- Looking for sin in m - found
-- Checking if LTO is supported
-- Checking if LTO is supported - no
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_INSTALL_LIBDIR
    CMAKE_USE_SYSTEM_LIBRARIES
    DOXYGEN_EXECUTABLE


-- Build files have been written to: /home/builder/.termux-build/rgbds/build
[1/62] Building C object src/CMakeFiles/rgbfix.dir/error.c.o
[2/62] Building C object src/CMakeFiles/rgbfix.dir/extern/getopt.c.o
[3/62] Building C object src/CMakeFiles/rgbfix.dir/_version.c.o
[4/62] Building C object src/CMakeFiles/rgbfix.dir/fix/main.c.o
[5/62] Linking C executable src/rgbfix
[6/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/main.cpp.o
/home/builder/.termux-build/rgbds/src/src/gfx/main.cpp:346:31: warning: result of comparison of constant 256 with expression of type 'std::array<unsigned char, 2>::value_type' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.baseTileIDs[0] >= 256) {
                            ~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
/home/builder/.termux-build/rgbds/src/src/gfx/main.cpp:362:31: warning: result of comparison of constant 256 with expression of type 'std::array<unsigned char, 2>::value_type' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.baseTileIDs[1] >= 256) {
                            ~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~
/home/builder/.termux-build/rgbds/src/src/gfx/main.cpp:477:27: warning: result of comparison of constant 256 with expression of type 'uint8_t' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
                        if (options.nbPalettes > 256) {
                            ~~~~~~~~~~~~~~~~~~ ^ ~~~
3 warnings generated.
[7/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/pal_sorting.cpp.o
[8/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/pal_packing.cpp.o
[9/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/pal_spec.cpp.o
[10/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/proto_palette.cpp.o
[11/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/process.cpp.o
[12/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/reverse.cpp.o
[13/62] Building C object src/CMakeFiles/rgbgfx.dir/extern/getopt.c.o
[14/62] Building C object src/CMakeFiles/rgbgfx.dir/error.c.o
[15/62] Building C object src/CMakeFiles/rgbgfx.dir/_version.c.o
[16/62] Building C object src/CMakeFiles/rgblink.dir/link/assign.c.o
[17/62] Building CXX object src/CMakeFiles/rgbgfx.dir/gfx/rgba.cpp.o
[18/62] Linking CXX executable src/rgbgfx
[19/62] Building C object src/CMakeFiles/rgblink.dir/link/main.c.o
[20/62] Building C object src/CMakeFiles/rgblink.dir/link/object.c.o
/home/builder/.termux-build/rgbds/src/src/link/object.c:670:14: warning: result of comparison of constant 1152921504606846975 with expression of type 'unsigned int' is always false [-Wtautological-constant-out-of-range-compare]
        if (nbFiles > SIZE_MAX / sizeof(*nodes))
            ~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
[21/62] Building C object src/CMakeFiles/rgblink.dir/link/output.c.o
[22/62] Building C object src/CMakeFiles/rgblink.dir/link/patch.c.o
[23/62] Building C object src/CMakeFiles/rgblink.dir/link/script.c.o
[24/62] Building C object src/CMakeFiles/rgblink.dir/link/section.c.o
[25/62] Building C object src/CMakeFiles/rgblink.dir/link/symbol.c.o
[26/62] Building C object src/CMakeFiles/rgblink.dir/extern/utf8decoder.c.o
[27/62] Building C object src/CMakeFiles/rgblink.dir/hashmap.c.o
[28/62] Building C object src/CMakeFiles/rgblink.dir/linkdefs.c.o
[29/62] Building C object src/CMakeFiles/rgblink.dir/opmath.c.o
[30/62] Building C object src/CMakeFiles/rgblink.dir/link/sdas_obj.c.o
[31/62] Building C object src/CMakeFiles/rgblink.dir/error.c.o
[32/62] Building C object src/CMakeFiles/rgblink.dir/_version.c.o
[33/62] Building C object src/CMakeFiles/rgblink.dir/extern/getopt.c.o
[34/62] Linking C executable src/rgblink
[35/62] Building C object test/CMakeFiles/randtilegen.dir/gfx/randtilegen.c.o
[36/62] Linking C executable test/randtilegen
[37/62] [BISON][PARSER] Building parser with bison 3.8.2
[38/62] Building CXX object test/CMakeFiles/rgbgfx_test.dir/gfx/rgbgfx_test.cpp.o
FAILED: test/CMakeFiles/rgbgfx_test.dir/gfx/rgbgfx_test.cpp.o 
/home/builder/.termux-build/_cache/android-r25b-api-24-v0/bin/clang++ --target=aarch64-none-linux-android --gcc-toolchain=/home/builder/.termux-build/_cache/android-r25b-api-24-v0 --sysroot=/home/builder/.termux-build/_cache/android-r25b-api-24-v0/sysroot -D_ISOC11_SOURCE -D_POSIX_C_SOURCE=200809L -I/home/builder/.termux-build/rgbds/src/include -I/data/data/com.termux/files/usr/include/libpng16 -fstack-protector-strong -Oz --target=aarch64-linux-android24  -I/data/data/com.termux/files/usr/include -O3 -DNDEBUG -fPIE -Wall -pedantic -std=gnu++17 -MD -MT test/CMakeFiles/rgbgfx_test.dir/gfx/rgbgfx_test.cpp.o -MF test/CMakeFiles/rgbgfx_test.dir/gfx/rgbgfx_test.cpp.o.d -o test/CMakeFiles/rgbgfx_test.dir/gfx/rgbgfx_test.cpp.o -c /home/builder/.termux-build/rgbds/src/test/gfx/rgbgfx_test.cpp
/home/builder/.termux-build/rgbds/src/test/gfx/rgbgfx_test.cpp:14:11: fatal error: 'spawn.h' file not found
        #include <spawn.h>
                 ^~~~~~~~~
1 error generated.
[39/62] Building C object src/CMakeFiles/rgbasm.dir/asm/parser.c.o
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2022-10-03 00:37:30 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/3170671047">3170671047</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/rgbds/build.sh
+++ b/packages/rgbds/build.sh
@@ -2,8 +2,12 @@ TERMUX_PKG_HOMEPAGE=https://rgbds.gbdev.io
 TERMUX_PKG_DESCRIPTION="Rednex Game Boy Development System - An assembly toolchain for the Nintendo Game Boy & Game Boy Color"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=0.5.2
+TERMUX_PKG_VERSION=0.6.0
 TERMUX_PKG_SRCURL=https://github.com/gbdev/rgbds/releases/download/v${TERMUX_PKG_VERSION}/rgbds-${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=142479e1b68e3947af2dace884fb0efa65e789163cb5909bdcbae1f132dc1457
+TERMUX_PKG_SHA256=a5fa40b92e0562c6a092bc81ca56614316730aca7b10b8177285023781ff0e32
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="libpng"
+TERMUX_PKG_DEPENDS="libpng, libandroid-spawn"
+
+termux_step_pre_configure() {
+	export LDFLAGS+=" -landroid-spawn"
+}

